MapboxMobileEvents/MMEMetricsManager.m

```diff
@@ -182,30 +247,75 @@ - (NSDictionary *)attributes {
     attributes[MMEEventSDKIdentifier] = [MMEEventsManager sharedManager].apiClient.userAgentBase;
     attributes[MMEEventSDKVersion] = [MMEEventsManager sharedManager].apiClient.hostSDKVersion;
     attributes[MMEEventKeyUserAgent] = [MMEEventsManager sharedManager].apiClient.userAgent;
-    
+
     return attributes;
 }
 
+- (MMEEvent *)loadPendingTelemetryMetricsEvent {
+    MMEEvent* pending = nil;
+
+    if ([NSFileManager.defaultManager fileExistsAtPath:MMEMetricsManager.pendingMetricsEventPath]) {
+        @try {
+            NSData *thenData = [NSData dataWithContentsOfFile:MMEMetricsManager.pendingMetricsEventPath];
+            NSKeyedUnarchiver* unarchiver = [NSKeyedUnarchiver.alloc initForReadingWithData:thenData];
+            unarchiver.requiresSecureCoding = YES;
+            pending = [unarchiver decodeObjectOfClass:MMEEvent.class forKey:NSKeyedArchiveRootObjectKey];
+        }
+        @catch (NSException *exception) {
+            [MMEEventLogger.sharedLogger logEvent:[MMEEvent debugEventWithException:exception]];
+        }
+    }
+
+    return pending;
+}
+
 - (MMEEvent *)generateTelemetryMetricsEvent {
     NSDate *zeroHour = [self.metrics.recordingStarted mme_startOfTomorrow];
-    if (zeroHour.timeIntervalSinceNow > 0) {
-        NSString *debugDescription = [NSString stringWithFormat:@"TelemetryMetrics event isn't ready to be sent; waiting until %@ to send", zeroHour];
+    NSString *metricsDate = [MMEDate.iso8601DateFormatter stringFromDate:NSDate.date];
+    MMEEvent *telemetryMetrics = [MMEEvent telemetryMetricsEventWithDateString:metricsDate attributes:self.attributes];
```

Comment:
If we'll be generating the telemetry event more often then we need to add the same logic for the zeroHour to eventsManager `sendTelemetryMetricsEvent:` or else the event will sent more often than 24 hour chunks.

Reply:
`sendTelemetryMetricsEvent` logic was faulty in this way -- should have had the conditional there in the first place I think.

Reply:
`generateTelemetryMetricsEvents` doesn't return the event if it's not time, it just writes the pending event and returns nil so that `sendTelemetryMetrics` doesn't need to know about scheduling

Reply:
Ahh, you're right -- sorry about that. Thanks for the walkthrough.